[Phylicia Christel] iP

phyliciachristel opened the pull request:
# Duke
> "Great and simple to use!" _- a trusty consumer_

Duke helps you make things easier, it's
- simple
- text-based
- ~~easy~~ very easy to use!

To start, just
1. download it from [here](https://github.com/phyliciachristel/ip)
2. add your tasks
3. and use it 🎉 

Features:
- [x] Managing tasks
- [ ] Reminders (coming soon)
- [ ] Handling deadlines (coming soon)

If you are a java programmer, you can check out the OOP used, here's the `main` method for example:
```ruby
public static void main(String[] args) {
    new Duke().run();
}
```

Commit:
Add Gradle support

Commit:
implemented level-1

Commit:
implemented level-2

Commit:
implemented Level-3

Commit:
implemented level-4

Commit:
removed unneeded comments

Commit:
implemented A-TextUiTesting

Commit:
implemented level-5

Commit:
implemented level-6

Commit:
added comments and introduced an addTask function

Commit:
implement level-7

Commit:
implement Level-8

Commit:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

Commit:
resolve conflict between merges

Commit:
- implement basic ui class functions
- add parser class and tasklist class
- add storage class with draft save()

Commit:
implement TaskList class
implement Storage class

Commit:
- Change classes and variables to non-static
- Implement readCommand()
- Write draft for Parser class using comments

Commit:
Implement basic Command class structure

Commit:
Implement Parser class before incorporating exceptions

Commit:
Implement A-Exceptions

Commit:
Add time input for event and finish up A-MoreOOP

Commit:
Implement A-Packages
- Create class file in package for each class used
- Split subclasses into their own class file

Commit:
Remove unnecessary comments and change wildcard import

Commit:
Merge branch 'add-gradle-support'

Commit:
Add JUnit tests for Todo and DeleteCommand classes

Commit:
Rename main Duke code and add to package

Commit:
Use relative filename

Commit:
Implement A-JavaDoc

Commit:
Implement A-CodingStandard

Commit:
Implement Level-9

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/AddCommand.java
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/Event.java
#	src/main/java/duke/MarkCommand.java
#	src/main/java/duke/UnmarkCommand.java

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

Commit:
Fix minor warnings

Commit:
Make code more concise

Commit:
Implement Level-10

czhongwei commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,88 @@
+    /**
+     * Constructor for the Ui class.
+     */
+    public Ui () {
```
I believe the constructor could be omitted if there is no body for it

czhongwei commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,88 @@
+     * @return String showing tasks with matching keyword.
+     */
+    public String showMatchingTasks(TaskList taskList) {
+        if(taskList.getSize() > 0) {
```
you could include a blank space before your if-condition to fit with the coding standard

seox123 replied:
I like this suggestion, and see a few other places where this issue occurs as well.

phyliciachristel replied:
thank you i actually completely forgot about this haha

czhongwei reviewed (commented):
Good work! just be mindful of including blank spaces before if-conditions!

csheun commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,62 @@
+     * {@inheritDoc}
+     */
+    @Override
+    public String formatToSave() {
```
Perhaps can have a naming leaning more towards action verb.

phyliciachristel replied:
thanks for your comments in the other parts of the code! however im pretty sure format is a verb haha :]

csheun commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import java.io.File;
```
I like how you pay attention to list the imports in lexicographical order.

csheun commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    /**
+     * Displays the total amount of tasks
+     * in the taskList.
```
```suggestion
     * Displays the total amount of tasks in the taskList.
     * 
```
I think you can consider shifting the second line up.

csheun commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,55 @@
+
+    private Duke duke;
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/shef kirbo.png"));
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/kirbo.png"));
```
I like how you put in effort to personalise your bot! Looks cute!

seox123 replied:
I like the customization using kirbo as well, it is very unique and cute!

csheun reviewed (commented):
Overall LGMT.
Appreciate the effort to customise your bot and making it really personal!

seox123 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,139 @@
+ */
+public class Parser {
+
+    /**
+     * Constructor for the Parser class.
+     */
+    public Parser() {
+
+    }
```
Perhaps consider leaving out the constructor if it is not necessary?

seox123 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,125 @@
+    @Override
+    public String toString() {
+        String str = "";
+        for(int i = 1; i <= this.getSize(); i++) {
+            Task task = this.getTask(i - 1);
+            str += i + "." + task.toString() + "\n";
+        }
```
Repeated string concatenation could be costly. Perhaps you could consider making use of a StringBuilder object?

phyliciachristel replied:
I did not think about that, thanks for the suggestion!

seox123 commented on src/main/java/duke/UnmarkCommand.java:
```diff
@@ -0,0 +1,41 @@
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) {
+
```
Perhaps you could remove this empty new line if it is unnecessary? I noticed the same issue in several other places too.

Commit:
Fix code with recommendations from peer review.

Commit:
Error for files are not printed

The user will not know if there is an eror in file saving.

Let's print the error messages whenever a change to the tasklist is
made and the file cannot be saved.

Commit:
Code does not use assertions

Assertions define assumptions so that the runtime can verify them.

Let's add some assertions in the code.

Using assertions ensure important assumptions are documented.

Commit:
Code has long methods and deep nesting

Code quality should be better.

Let's break down parse method into different methods to parse
each specific command.

This makes the code more readable and adheres to the coding
quality guidelines.

phyliciachristel committed:
Merge pull request #1 from phyliciachristel/branch-A-Assertions

Branch a assertions

phyliciachristel committed:
Merge branch 'master' into branch-A-CodeQuality

phyliciachristel committed:
Merge pull request #2 from phyliciachristel/branch-A-CodeQuality

Branch A-CodeQuality

Commit:
Implement C-Archive

Commit:
- Change date display format for Deadline events
- Remove unused methods
- Add a new Show command
- Add Ui.png

Commit:
Separate classes further into a command and task package

Commit:
Add User Guide and edit display

Commit:
Fix small formatting details and update UI image

Commit:
Remove deadcode, update User guide and edit formatting

Commit:
Fix UI format errors and edit build gradle file
